src/Entity/ProductCommentCriterion.php

```diff
@@ -108,4 +201,17 @@ public function setActive($active)
 
         return $this;
     }
+
+    /**
+     * @return bool
+     */
+    public function isValid()
+    {
+        $res = true;
+        foreach ($this->names as $key => $value) {
+            $res &= Validate::isGenericName($value);
+        }
+
+        return $res;
+    }
```

Comment:
```suggestion
    public function isValid()
    {
        foreach ($this->getNames() as $value) {
            if(!Validate::isGenericName($value)) {
                return false;
            }
        }

        return true;
    }
```

Reply:
Performances improvement :)

Reply:
@leemyongpakvn You don't need to loop all the names... just have a false for the condition to return false

Reply:
@PululuK You're right. Applied :)